File: python/diagonal_negativos.py
n = int(input("Qual a ordem da matriz? "))

matriz: [[int]] = [[0 for x in range(n)] for x in range(n)]

for i in range(0, n):
    for j in range(0, n):
        matriz[i][j] = int(input(f"Elemento [{i},{j}]: "))

negativos = 0
print("DIAGONAL PRINCIPAL:")
for i in range(0, n):
    for j in range(0, n):
        if i == j:
            print(f"{matriz[i][j]} ", end="")
        if matriz[i][j] < 0:
            negativos += 1

print()
print(f"QUANTIDADE DE NEGATIVOS = {negativos}")

File: java/diagonal_negativos.java
import java.util.Scanner;

public class Main {

	public static void main(String[] args) {
		
		Scanner sc = new Scanner(System.in);
		
		int N, i, j;
		
		System.out.print("Qual a ordem da matriz? ");
		N = sc.nextInt();
		
		int[][] matriz = new int[N][N];
		
		for(i = 0; i < N; i++) {
			for(j = 0; j < N; j++) {
				System.out.print("Elemento [" + i +"," + j + "]: ");
				matriz[i][j] = sc.nextInt();
			}
		}
		
		System.out.println("DIAGONAL PRINCIPAL:");
		int negativos = 0;
		
		for(i = 0; i < N; i++) {
			for(j = 0; j < N; j++) {
				if (i == j) {
					System.out.print(matriz[i][j] + " ");
				}
				if (matriz[i][j] < 0) {
					negativos++;
				}
			}
		}
		
		System.out.println();
		System.out.print("QUANTIDADE DE NEGATIVOS = " + negativos);
		
	sc.close();
	}
	
}
